### PR TITLE
LibGUI+LibGfx: Invert (solid color) button icons with low contrast ratios

### LINKED ISSUE #13978: Dark icons become hard/impossible to see with dark themes
image

Here's an example with in Pixel Paint, though I expect there's other places where this is the case too.
I can barely see the undo/redo, line tool, rectangle tool, or ellipse tool (+ some other buttons)

Two solutions I can think of:
 - Simply to draw white outlines around these icons.
- Automatically invert the icons if the contract ratio becomes too low.

### PR DESCRIPTION
**Pull request by MacDue**
On some dark themes, it becomes impossible to see dark button icons against their dark button backgrounds. 
This change tries to mitigate that by inverting the icon color if the contrast ratio (against the button background) is less the 4.5 (the recommended minimum for text).

This is only done for icons that are a solid color (e.g. all back), where the desired icon would likely be the same inverted anyway.

**Demo:**
(Look at the undo/redo buttons, along with the line, ellipse, and rectangle tools)

https://user-images.githubusercontent.com/11597044/167319953-5880e1c5-36c9-4817-abe9-01b05ee37a6e.mp4

Before these changes the icons looked like this:

image
^ the line, rectangle, and ellipse tools icons before (barely visible) 

I've made these changes to the base `GUI::Button` class as there's lots of apps all over the system where this applies. 

(Fixes a lot of cases of https://github.com/SerenityOS/serenity/issues/13978)

**Note:** This PR should not close #13978 entirely there's still more work to be done (some more complex icons will still need dark variants)

### CODE REVIEW

**Comment by djwisdom** [comment]
This is good™.  :^)

File: Userland/Libraries/LibGfx/Bitmap.cpp [hunk 1]
```diff
@@ -459,6 +459,17 @@ ErrorOr<NonnullRefPtr<Gfx::Bitmap>> Bitmap::cropped(Gfx::IntRect crop) const
     return new_bitmap;
 }
 
+ErrorOr<NonnullRefPtr<Gfx::Bitmap>> Bitmap::inverted() const
+{
+    auto new_bitmap = TRY(clone());
```
Comment (Hendiadyoin1) [inline Userland/Libraries/LibGfx/Bitmap.cpp:464]: I think `clone` is a bit overkill here,
You are not using the stored data and are just overriding it directly, so
we can save a memcpy by just doign a
```suggestion
    auto new_bitmap = TRY(Bitmap::try_create(format(), size(), scale()));
```

Comment (MacDue) [reply Userland/Libraries/LibGfx/Bitmap.cpp:464]: Ah okay. Just thinking for this PR I think any allocations at all could be avoided if this method was changed to `invert()` to invert the pixels in-place, then just invert them again after painting.

File: Userland/Libraries/LibGfx/Bitmap.cpp [hunk 2]
```diff
@@ -469,6 +469,15 @@ ErrorOr<NonnullRefPtr<Bitmap>> Bitmap::to_bitmap_backed_by_anonymous_buffer() co
     return bitmap;
 }
 
+void Bitmap::invert()
+{
+    for (auto y = 0; y < height(); y++) {
+        for (auto x = 0; x < width(); x++) {
+            set_pixel(x, y, get_pixel(x, y).inverted());
+        }
```
Comment (linusg) [inline Userland/Libraries/LibGfx/Bitmap.cpp:477]: no curly braces needed here

**Review by linusg — COMMENTED** [review]
This is very nice! Tiny code style nitpick.

File: Userland/Libraries/LibGfx/Bitmap.cpp (hunk 2 above)
Comment (MacDue) [reply Userland/Libraries/LibGfx/Bitmap.cpp]: :laughing: you ninja-ed me, I had this open in a tab & was just about to post in #code_review then this popped up

**Comment by djwisdom** [comment]
This is just wonderful. :^) Merged at last!
At least now there's a fallback should the Durrque Icon Theme pack is a no-go.
Snapshot:
image

**Pull request merged**